pkg/datastore/mysql/mysql.go

```diff
@@ -82,22 +84,180 @@ func (m *MySQL) Find(ctx context.Context, kind string, opts datastore.ListOption
 
 // Get implementation for MySQL
 func (m *MySQL) Get(ctx context.Context, kind, id string, v interface{}) error {
-	return datastore.ErrUnimplemented
+	row := m.client.QueryRowContext(ctx, buildGetQuery(kind), id)
+	var val string
+	err := row.Scan(&val)
+	if err == sql.ErrNoRows {
+		return datastore.ErrNotFound
+	}
+	if err != nil {
+		m.logger.Error("failed to get entity",
+			zap.String("id", id),
+			zap.String("kind", kind),
+			zap.Error(err),
+		)
+		return err
+	}
+
+	return decodeJSONValue(val, v)
 }
 
 // Create implementation for MySQL
 func (m *MySQL) Create(ctx context.Context, kind, id string, entity interface{}) error {
-	return datastore.ErrUnimplemented
+	stmt, err := m.client.PrepareContext(ctx, buildCreateQuery(kind))
+	if err != nil {
+		m.logger.Error("failed to create entity: failed to prepare query",
+			zap.String("id", id),
+			zap.String("kind", kind),
+			zap.Error(err),
+		)
+		return err
+	}
+	defer stmt.Close()
+
+	data, err := encodeJSONValue(entity)
+	if err != nil {
+		m.logger.Error("failed to create entity: failed to encode json data",
+			zap.String("id", id),
+			zap.String("kind", kind),
+			zap.Error(err),
+		)
+		return err
+	}
+
+	// In case the given model is `project`, id is not in uuid type so just generate random uuid instead.
+	if kind == "Project" {
+		id = uuid.New().String()
```

Comment:
I just realized that this causes a problem for the Project table, that is ErrAlreadyExists error will never be returned because the ID is always newly created. So we still need the hashing.

Reply:
You're right 👍 Looks like we should drop generate random UUID and make all `table id` from `model.id`. Will address in another PR 🙏

Reply:
addressed on https://github.com/pipe-cd/pipe/pull/1658 🙏